File: src/main/java/core/GenomePerspective.java
package core;

import java.io.File;
import java.io.IOException;
import java.sql.SQLException;

import core.utils.Genome;
import core.utils.Perspectives;

/**
 * The genomic perspective on a clone.
 * @author noemi
 *
 */
public class GenomePerspective extends Perspective{

	public GenomePerspective(float f, String sampleName, String[] nMut) throws SQLException {
		super(f, sampleName);
		this.profile=new Genome(nMut);
	}
	
	public GenomePerspective(File outFile,String rootName) throws Exception {
		super(outFile,Perspectives.GenomePerspective, rootName);
	}

	@Override
	public Clone getPerspective(Perspectives whichP) {
		if(whichP.equals(Perspectives.valueOf(this.getClass().getSimpleName()))){
			return this;
		}
		return null;
	}


}

File: webportal/frontend/src/lib/cloneid/core/GenomePerspective.js
import {Genome} from './utils/Genome';
import {Perspectives} from './utils/Perspectives';
import {Perspective} from './Perspective';

export class GenomePerspective extends Perspective {

    /**
     * @param {number} f
     * @param {string} sampleName
     * @param {string[]} nMut
     */
    constructor(f, sampleName, nMut) 
     {
        super(f, sampleName);
        this.profile = new Genome(nMut);
    }

    /**
     * @param {Perspectives} whichP
     */
    getPerspective(whichP) {
        if (whichP.equals(Perspectives.valueOf(this.constructor.name))) {
            return this;
        }
        return null;
    }
}
